doc/source/data/transforming-datasets.rst

```diff
@@ -373,6 +373,97 @@ The following output types are allowed for per-row UDFs (e.g.,
     :start-after: __writing_simple_out_row_udfs_begin__
     :end-before: __writing_simple_out_row_udfs_end__
 
+.. _datasets-groupbys:
```

Comment:
Shall we put this section to the end of this file?

`Writing User-defined Functions (UDFs)` -> `Configuring Batch Size` -> `Compute Strategy` is a coherent story for writing `map_batches`.